src/entities/friendship_history.rs

```diff
@@ -134,6 +142,43 @@ impl FriendshipHistoryRepository {
         }
     }
 
+    /// Fetches the request events of the given user.
+    pub async fn get_user_pending_request_events(
+        &self,
+        address: &str,
+    ) -> Result<Vec<FriendshipRequestEvents>, sqlx::Error> {
+        let query = USER_REQUESTS_QUERY.to_string();
+
+        let query = sqlx::query(&query).bind(address.to_ascii_lowercase());
+
+        let executor = self.get_executor(None);
+
+        let (res, _) = DatabaseComponent::fetch_all(query, executor).await;
+
+        match res {
+            Ok(rows) => {
+                let response = Ok(rows
+                    .iter()
+                    .map(|row| -> FriendshipRequestEvents {
+                        FriendshipRequestEvents {
+                            address_1: row.try_get("address_1").unwrap(),
+                            address_2: row.try_get("address_2").unwrap(),
+                            acting_user: row.try_get("acting_user").unwrap(),
+                            timestamp: row.try_get("timestamp").unwrap(),
+                            metadata: row.try_get("metadata").unwrap(),
+                        }
```

Comment:
**Suggestion:** This might be better suited for a more general refactor and not this particular PR, but we might be able to avoid this kind of parsing by deriving `sqlx::FromRow` in the struct and have sth like the following:
```rust
use sqlx::FromRow;

#[derive(FromRow)]
pub struct FriendshipRequestEvents {
  // all fields
}

// ...

let row: PgRow = ...;
let friendship_request_events = FriendshipRequestEvents::from_row(row).expect("to be a friendship request events");
```

Reply:
Uuuh nice, I'll update this PR for this specific query, and I'll create a ticket to review the other queries at a later time